### DISCUSSION TITLE
Synchronizing Bonvision stimuli

### DISCUSSION THREAD

**Discussion by bruno-f-cruz** [bonsai-rx/bonsai#885] [Q&A] [answered]
The following question was asked in the Bonsai Users - Google Groups
[Link to original post](https://groups.google.com/g/bonsai-users/c/d2ANyYNnBUg)

---
**Original post**

I am trying to make a synchronisation square appear in the corner of the screen alternating between black and white for the duration of each stimulus presentation.

I tried various combinations of Multicast and behavioural subject, but nothing seems to work, ranging between not doing anything or crashing.

I am attaching my latest try. I would appreciate any suggestions.

Cheers
Liad

**Comment by bruno-f-cruz** [comment, marked as answer]
Hi Liad, 

I left two options coded in the attached workflow that should get you started. One relies on coding the quad in the observable that generates the stimuli and the second one on sending a boolean to a 'DrawQuad' outside the stimuli generation workflow. 

Let meknow if you have any question regarding the workflow!
[syncExample.zip](https://github.com/bonsai-rx/bonsai/files/8557441/syncExample.zip)

**Reply by liadJB** [reply]
Hey Bruno, 
Thanks for the response.
What I am looking for is slightly different. 
I want the square to appear either white or black on stimulus presentation (there is no ISI), so on every other stim presentation the square is either black or white. If you can have any suggestion on how to go about that it'd be great.

Thanks alot,

Liad

**Reply by bruno-f-cruz** [reply]
![image](https://user-images.githubusercontent.com/7049351/165336231-e74f4139-99c3-44c9-888a-2d2ae2c7638c.png)
Try this modification
`Accumulate` -> `Mod(2)`
Additionally, set the bottom double to 0.5 so it draws a gray screen where there is nothing there. You can change this example to not include the ITI (or set the timer to 0) but i want to leave it in for future reference :)
cheers,
B
